### ISSUE TITLE
Unauthenticated uploads (e.g. via source) do not trigger `job-done` event

### ISSUE THREAD

**Issue by madduck** [eikek/docspell#2342]
If I upload an item via a source, no `job-done` notification event is triggered.

Looking at debug/trace logs, the following appears curious:

```
docspell.store.impl.DoobieMeta - SQL: Success(SELECT c.id, c.name, u.uid, u.login, u.password, c.state, u.state, u.account_source FROM user_ u  INNER JOIN collective c  ON u.coll_id = c.id WHERE (c.id = ?  AND u.login = c.name),List(CollectiveId(1)),804277 nanoseconds,80443 nanoseconds)

docspell.scheduler.impl.SchedulerImpl.makeJobDoneEvent:246 - Found job owner None for job RJob(Ident(EmTTECm8YB2-WD8wSCfMvoq-pTzQr3wBozp-KXwPAq9Esjq),Ident(process-item),Ident(1),{"meta":{"collective":1,"itemId": …
```

There is no job owner, unless maybe the username and collective are the same (cf. the SQL). I have not tested that, but I did verify that the even fires just fine as soon as I do an upload via `sec/upload/item` instead.

**Comment by eikek** [comment]
Yes, this needs to get a deeper look. The notification channels are maintained for each user. So if there is no user, we could probably simply notify all users that have a corresponding channel defined.

**Comment by madduck** [comment]
It makes sense to have per-user notification channels (sinks), but not really per-user events (sources). If a colleague of mine uploads a new document, given that I can access it due to the absence of users/permissions, then I should be able to receive the associated event (source) in the way I want (sink).

The Matrix integration is great, but in a team, there is one person that is just slightly less interested in receiving a notification about an upload than everyone else, and that is the person making the upload… for that use-case, there really ought to be per-collective notification channels…

**Comment by eikek** [comment]
right, it needs a deeper look - I can't remember why it is like this now (I don't think only the "owner" is notified, have to look into the code…). As I wrote, I think we can simply go through all users of  the collective and notify every one that has a corresponding channel configured. not sure, why but it seems it's not the case currently
